common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1636,14 +1699,13 @@ public int levenshteinDistance(UTF8String other, int threshold) {
         } else if (i == min - 1) {
           num_bytes_i = 0;
         } else {
-          if (ByteArrayMethods.arrayEquals(t.base, t.offset + j_bytes,
-                  s.base, s.offset + i_bytes, num_bytes_j)) {
+          num_bytes_i = numBytesForFirstByte(s.getByte(i_bytes));
```

Comment:
I don't quite get the details here. Why do we need to pre-evalute `numBytesForFirstByte` now? It was evaluated at the end of the loop body before.